### PR TITLE
Tests some event edge cases

### PR DESCRIPTION
**Pull request by alpe**
Add more tests for empty or nil events or event attributes

### CODE REVIEW

File: x/wasm/keeper/events_test.go [hunk 1]
```diff
@@ -183,6 +183,13 @@ func TestNewCustomEvents(t *testing.T) {
 				sdk.NewAttribute("_contract_address", myContract.String()),
 				sdk.NewAttribute("my Key", "myVal"))},
 		},
+		"empty event elements": {
+			src:     make(wasmvmtypes.Events, 10),
+			isError: true,
+		},
+		"nil": {
+			exp: sdk.Events{},
+		},
```
Comment (pinosu) [inline x/wasm/keeper/events_test.go:192]: 👍

File: x/wasm/keeper/events_test.go [hunk 2]
```diff
@@ -240,6 +247,15 @@ func TestNewWasmModuleEvent(t *testing.T) {
...
 		},
+		"empty elements": {
+			src:     make([]wasmvmtypes.EventAttribute, 10),
+			isError: true,
+		},
+		"nil": {
+			exp: sdk.Events{sdk.NewEvent("wasm",
+				sdk.NewAttribute("_contract_address", myContract.String()),
+			)},
+		},
```
Comment (pinosu) [inline x/wasm/keeper/events_test.go:258]: 👍

File: x/wasm/keeper/gas_register_test.go [hunk 3]
```diff
@@ -283,6 +283,26 @@ func TestReplyCost(t *testing.T) {
...
+		},
+		"subcall response with events unset": {
+			src: wasmvmtypes.Reply{
+				Result: wasmvmtypes.SubcallResult{
+					Ok: &wasmvmtypes.SubcallResponse{},
+				},
+			},
+			srcConfig: DefaultGasRegisterConfig(),
+			exp:       DefaultInstanceCost,
+		},
```
Comment (pinosu) [inline x/wasm/keeper/gas_register_test.go:305]: 👍

File: x/wasm/keeper/gas_register_test.go [hunk 4]
```diff
@@ -298,6 +318,33 @@ func TestReplyCost(t *testing.T) {
...
+	}
+	for name, spec := range specs {
+		t.Run(name, func(t *testing.T) {
+			gotGas := NewDefaultWasmGasRegister().EventCosts(spec.srcAttrs, spec.srcEvents)
+			assert.Equal(t, spec.expGas, gotGas)
+		})
+	}
+}
+
```
Comment (pinosu) [inline x/wasm/keeper/gas_register_test.go:347]: good test

Comment (pinosu) [inline x/wasm/keeper/gas_register_test.go:347]: good test

**Review by pinosu — APPROVED** [review]
lgtm 👍
